### DISCUSSION TITLE
怎样实现对PAG元素的自定义组装

### DISCUSSION THREAD

**Discussion by LUJYM** [Tencent/libpag#1618] [Q&A] [answered]
请问pag怎样实现我下面的这种功能：
比如一个pag文件里面是人物A在跳舞
另外一个pag文件是一顶帽子B
还有很多pag文件是服装、头发或者眼镜之类的
我可以通过合成composition这种方式或者动态替换image实现自定义组装特效里面的内容吗

**Comment by kevingpqi123** [comment, marked as answer]
都是支持的，关于组合使用，可以看下 demo 工程中多个 pag 文件放在一个 PAGView 中播放的范例，通过 setMatrix 设置位置信息，setStartTime 设置相对于主时间轴的开始时间，至于动态修改文本和替换占位图，和之前的使用方法一致
